From c7564527fc2800da59313c2a032c9a88ac360ddf Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <daniel.dragomir@windriver.com>
Date: Mon, 30 Apr 2018 12:24:50 +0300
Subject: [PATCH 269/278] tools/perf: Include inttypes.h to avoid error in
 cs-etm.c

PRIx64 macro was not defined and this error occours:

| util/cs-etm.c:1264:40: error: expected '}' before 'PRIx64'
|    [CS_ETM_MAGIC] = "   Magic number  %"PRIx64"\n",
	|                                         ^~~~~~

This was introduced by commit 362f8417b1cc ("opencsd: This is a
Backport of the Following") from axxia branches. This backport was
done in 4.9 Kernel where inttypes.h was included through
tools/perf/util/util.h.

After porting in 4.12 Kernel, this error occours because starting with
commit c3dca1a1c0ec ("perf tools: Remove PRI[xu] macros from perf.h"),
inttypes.h was not included in util.h.

So, include inttypes.h directly in cs-etm.c to fix this.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Daniel Dragomir <daniel.dragomir@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 tools/perf/util/cs-etm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/perf/util/cs-etm.c b/tools/perf/util/cs-etm.c
index d48b9a4..d34d7d5 100644
--- a/tools/perf/util/cs-etm.c
+++ b/tools/perf/util/cs-etm.c
@@ -37,6 +37,7 @@
 #include "debug.h"
 
 #include <stdlib.h>
+#include <inttypes.h>
 
 #define KiB(x) ((x) * 1024)
 #define MiB(x) ((x) * 1024 * 1024)
-- 
1.8.5.2.233.g932f7e4

